src/openforms/forms/tests/test_migrations.py

```diff
@@ -435,3 +435,57 @@ def test_fix_badly_formatted_rules(self):
             self.rule_badly_formatted.json_logic_trigger,
             {"==": [{"var": "selectBoxes.a"}, True]},
         )
+
+
+class TestLogicParsing(TestMigrations):
+    migrate_from = "0002_auto_20210917_1114_squashed_0045_remove_formstep_optional"
+    migrate_to = "0046_convert_advanced_logic"
+    app = "forms"
+
+    def setUpBeforeMigration(self, apps):
+        self.FormDefinition = apps.get_model("forms", "FormDefinition")
+        self.components = [
+            {
+                "type": "fieldset",
+                "key": "fieldset-1",
+                "logic": [
+                    {
+                        "trigger": {
+                            "type": "simple",
+                            "simple": {
+                                "show": True,
+                                "when": "non-existing-object",
+                                "eq": "trigger value",
+                            },
+                        },
+                        "actions": [
+                            {
+                                "type": "property",
+                                "property": {
+                                    "label": "Hidden",
+                                    "value": "hidden",
+                                    "type": "boolean",
+                                },
+                                "state": False,
+                            }
+                        ],
+                    },
+                ],
+            },
+        ]
+        self.FormDefinition.objects.create(
+            name="Definition with textfield",
+            slug="definition-with-textfield",
+            configuration={"components": self.components},
+        )
+
+    def test_parse_trigger(self):
+        """Check that :func:`parse_trigger` does not error out on retrieval of form(definition)
+
+        The test checks that migrations from 1.1.x to 2.0.1 with forms containing
+        references to non-existing objects do not crash. The forms themselves are left
+        unmodified (garbage in, garbage out).
+        """
+
+        form_def = self.FormDefinition.objects.get()
+        self.assertEqual(form_def.configuration["components"][0]["key"], "fieldset-1")
```

Comment:
```suggestion
        FormDefinition = self.apps.get_model("forms", "FormDefinition")
        form_def = FormDefinition.objects.get()
        self.assertEqual(form_def.configuration["components"][0]["key"], "fieldset-1")
```

Then, to test the actual migration behaviour, you'll want to query for form logic records and assert none exist (as the creation is skipped due to the bad input logic).